### ISSUE TITLE
`tracer` release step by step

### ISSUE THREAD

**Issue by OlivierBBB** [Consensys/besu-sequencer-plugins#999]
# Steps to doing a release of the `tracer`

After @powerslider. The goal is to produce a sharable snapshot + tag for the latest version of `arith-dev`.

## Terminal

```bash
// go to the root of the arithmetization repository
git switch arith-dev
git fetch origin
git pull
```
Create a release branch; `<release-branch-name>` is of the form `rel/0.x.y-rcz` (`rc` stands for release candidate) or `rel/0.x.y-beta`

```bash
git checkout -b <release-branch-name>
```
To simplify matters I will use `rel/0.x.y-rcz ` for the remainder of this

Open `gradle.properties` and replace the `releaseVersion` (no `rel/`)
```bash
releaseVersion=0.X.Y-rcZ     =>     releaseVersion=0.x.y-rcz
```
Modify the `CHANGELOG.md` file: `git log` to find the previous release in the commit history, copy all the commit messages until the most recent and open `CHANGELOG.md`, then insert them to the top of the file and format things nicely (and remove signatures, vacuous commits etc ...) **Note.** The links have to be added manually, but it's sufficient to grab one and to modify the PR number at the 2 places where it occurs.

Here is an example. Old `CHANGELOG.md`
```markdown
# Changelog

## 0.5.0-beta
* Merged the HUB branch ([#748](https://github.com/Consensys/linea-tracer/pull/748))

...
```

New `CHANGELOG.md`
```markdown
# Changelog

## 0.5.1-beta
* fix: removed old documentation files ([#995](https://github.com/Consensys/linea-tracer/pull/995))
* feat(txndata): implement txn data update handling the max nonce ([#987](https://github.com/Consensys/linea-tracer/pull/987))
* feat(hub): fill pMiscOobData9 ([#985](https://github.com/Consensys/linea-tracer/pull/985))
* feat(ecdata): use constants for call data sizes ([#983](https://github.com/Consensys/linea-tracer/pull/983))
* fix(hub): static exceptions that should not be thrown. ([#986](https://github.com/Consensys/linea-tracer/pull/986))
* feat(ecdata): switch from stacked set to stacked list ([#981](https://github.com/Consensys/linea-tracer/pull/981))
* fix(Makefile): updated file path to AccountFragment.java

## 0.5.0-beta
* Merged the HUB branch ([#748](https://github.com/Consensys/linea-tracer/pull/748))

...
```

Now add these changes
```bash
git add gradle.properties CHANGELOG.md
```

You should see something like this

<img width="627" alt="image" src="https://github.com/user-attachments/assets/919d9003-e8de-4bab-91fa-253837055dcf">

Commit and push
```
git commit -m "release: bump version to 0.x.y-rcz and update changelog"
git push --set-upstream origin rel/0.x.y-rcz
```

## Github

Go to [github](https://github.com/Consensys/linea-tracer) and create a PR. You can skip the tests if no modifications were made and tests were previously successful. Your files changed should look something like this
![image](https://github.com/user-attachments/assets/e6e9fb7b-2e63-410e-9989-40e5d9ed5c3b)

Merge into `arith-dev` and add your signature e.g.
```

Signed-off-by: Prénom Nom <prénom.nom@internet.web>
```

The click on `Restore branch`

![image](https://github.com/user-attachments/assets/60f5f0ad-89ce-4262-96f2-e8a57049cd21)

## Terminal

Go back to the terminal
```bash
git switch arith-dev
git pull
git log
```
You should see the release commit e.g.
<img width="709" alt="image" src="https://github.com/user-attachments/assets/4a86f297-827d-4421-956a-cb9a9b2e5918">

Create the tag --- **DON'T FORGET THE** `v` it appears **3** times below --- and push the tag:
```bash
git tag -a "v0.x.y-rcz" -m "release: v0.x.y-rcz"
git push -u origin "v0.x.y-rcz"
```
You should see

<img width="708" alt="image" src="https://github.com/user-attachments/assets/46a63e8b-5e73-4591-92f9-5fb992f32631">

## Githhub

Go to [Github Actions](https://github.com/Consensys/linea-tracer/actions) and you should see something like so

![image](https://github.com/user-attachments/assets/d350a859-be56-46e5-8447-0de3637ad97d)

and upon clicking on it

![image](https://github.com/user-attachments/assets/a8bae5e5-2178-4dcd-a70c-dad7ba936658)

Once that is done go to the [latest releases page](https://github.com/Consensys/linea-tracer/releases) by clicking on the Releases link on the repo home page

![image](https://github.com/user-attachments/assets/0d4b0c68-4f55-463a-b412-222e2de93cee)

Click on your release, Click on the edit button

![image](https://github.com/user-attachments/assets/00fcbdcf-7249-4fce-9a0c-f044e0a6add4)

Paste what you added to the `CHANGELOG.md` to the description and click on `Set as a pre-release` then click on `Publish release`.

![image](https://github.com/user-attachments/assets/cf0a1ef2-eaac-4e29-8b0d-6d6d31f09dee)

At this point @powerslider tells us to _click on edit again and to_ `Set as the latest release`. In the present case (I'm producing a beta release) I didn't do that step.

The final step is to share the release with the relevant people.

**Comment by letypequividelespoubelles** [comment]
... or just go on release page, and create a release ;)
